packages/frames.js/src/dscvr/index.ts

```diff
@@ -1,10 +1,28 @@
-import { FrameActionPayload } from '..';
+import {
+  type DscvrFramesRequest,
+  type DscvrClientProtocol,
+  type DscvrValidationResponse,
+  validateClientProtocol,
+  validateFramesPost,
+} from "@dscvr-one/frames-adapter";
+import { FrameActionPayload } from "..";
 
-export function isDscvrFrameActionPayload(
+export const isDscvrFrameActionPayload = (
   frameActionPayload: FrameActionPayload
-) {
+): frameActionPayload is DscvrFramesRequest => {
   return (
     !!frameActionPayload.clientProtocol &&
-    frameActionPayload.clientProtocol.startsWith('dscvr@')
+    validateClientProtocol(frameActionPayload.clientProtocol)
   );
-}
+};
+
+export const getDscvrFrameMessage = async (
```

Comment:
sorry for the back and forth on this, it's probably more appropriate for us to call this `validateDscvrFrameMessage`

Reply:
@ncpenke even though XMTP calls it `getXmtpFrameMessage`?

Reply:
I think their name doesn't make sense (IMO)

Reply:
The farcaster one calls it validate as well